frontend/app/hooks/use-authorization.tsx

```diff
@@ -0,0 +1,112 @@
+import { useSession, useUser } from "@clerk/react-router";
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+  type PropsWithChildren,
+} from "react";
+import { useLocation, useNavigate } from "react-router";
+import { BASE_URL } from "types/api";
+import type { MediamindUser } from "types/model";
+
+type UseAuthorizationReturn = {
+  sessionToken?: string;
+  isLoaded: boolean;
+  isSignedIn: boolean;
+  authorizationHeaders: Record<string, string>;
+  user?: MediamindUser;
+};
+
+const initialValue: UseAuthorizationReturn = {
+  isLoaded: false,
+  isSignedIn: false,
+  authorizationHeaders: {},
+  sessionToken: undefined,
+  user: undefined,
+};
+
+const AuthorizationContext = createContext(initialValue);
+
+// TODO: add mediamind backend request to return role/rights of user (+ within an organization)
+export const AuthorizationContextProvider = ({
+  children,
+}: PropsWithChildren) => {
+  const [token, setToken] = useState<string | undefined>();
+  const { isSignedIn, isLoaded } = useUser();
+  const { session } = useSession();
+  const [mediamindUser, setMediamindUser] = useState<MediamindUser>();
+
+  const navigate = useNavigate();
+  const { pathname } = useLocation();
+
+  useEffect(() => {
+    session?.getToken().then((t) => t && setToken(t));
```

Comment:
```suggestion
  const fetchToken = async () => {
    if (session) {
      const token = await session.getToken();
      if (token) {
        setToken(token);
      }
    }
  };
  fetchToken();
```

use async/await instead of ".then". ".then" is widely considered bad practice. there are some differences, but actually they are neglectable in modern JS programming and async/await is just cleaner